clustermesh-apiserver/metrics/metrics.go

```diff
@@ -36,7 +35,7 @@ type MetricsConfig struct {
 }
 
 func (def MetricsConfig) Flags(flags *pflag.FlagSet) {
-	flags.String(option.PrometheusServeAddr, def.PrometheusServeAddr, "Address to serve Prometheus metrics")
+	flags.String("prometheus-serve-addr", def.PrometheusServeAddr, "Address to serve Prometheus metrics")
```

Comment:
clustermesh-apiserver metrics are no longer exported, because they are no longer initialized: https://github.com/cilium/cilium/blob/a7a483ac3782574e56d63e7adbd5d6fb61686743/clustermesh-apiserver/metrics/metrics.go#L67-L73 I think it should be enough to make the metrics cell depend on the `LegacyMetrics` struct.

Reply:
Making `LegacyMetrics` a dependency requires that `metrics.Cell` is included in the clustermesh-apiserver hive, but that would cause a collision since they both register this same flag.

For now I solved it by just calling the constructor without hive. In the future, once these KVStore metrics have their own metric cell we can make the clustermesh-apiserver "just" re-use the `metrics.Registry` without the legacy metrics but include the KVStore metric cell in the hive. That way we have 1 copy of the registry logic but different hives include different metrics. The same goes for the operator metrics